tests/31sync/17peeking.pl

```diff
@@ -0,0 +1,175 @@
+use Future::Utils qw( repeat );
+
+# Tests MSC2753 style peeking
+
+test "Local users can peek into world_readable rooms by room ID",
+   requires => [ local_user_and_room_fixtures(), local_user_fixture() ],
+
+   check => sub {
+      my ( $user, $room_id, $peeking_user ) = @_;
+
+      matrix_set_room_history_visibility( $user, $room_id, "world_readable" )->then(sub {
+         matrix_send_room_text_message_synced( $user, $room_id, body => "something to peek");
+      })->then(sub {
+         do_request_json_for( $peeking_user,
+            method => "POST",
+            uri    => "/r0/peek/$room_id",
+            content => {},
+         )
+      })->then(sub {
+         matrix_sync( $peeking_user );
```

Comment:
This is racey. There's no guarantee that a `/sync` immediately after a 200 OK from `/peek` will result in the `peek` block existing, because it takes time for `syncapi` to consume Kafka events.

Reply:
hum, true. what's the right soln then? to retry until it turns up?

Reply:
i'm going to assume so; done. i'm a bit confused on why just setting a timeout on the sync isn't good enough though.

Reply:
> hum, true. what's the right soln then? to retry until it turns up?

Yes, there are helper functions to do this.